### PR TITLE
feat: production config for vaults

### LINKED ISSUE #7311: production bootstrap config for vaults
## What is the Problem Being Solved?

similar to...
 - #6284

but for vaults instead of pismo

similar to
 - #6957 

but for production instead of testing

## Description of the Design

```[tasklist]
- [x] #7431
- [x] #7467
- [x] get production IBC denom for ATOM
```

nice to have:
 - #7309

## Security Considerations

This fixes the addresses of EC members (again) and initial Oracle operators.

## Scaling Considerations

can't thin of any

## Test Plan

_IOU_

### PR DESCRIPTION
**Pull request by dckc**
closes: #7311

## Description
 - start with a copy of test-vaults-config
 - update comment at top; add to test-boot-config unit test
 - no endorsed ui
 - PSM configs: for USDC, USDT, DAI x axl, grv
 - econ committee size, addresses
 - Oracle addresses, minSubmissionCount
 - ATOM config: 0 mint limit, IBC denom, no initial price

todo
 - [x] upgrade handler
 - [x] require EC action before anyone can create a vault

### Security Considerations

 - it's the root of a software supply chain
 - includes addresses of accounts with oracle, EC authorities

### Scaling Considerations

All performance knobs are copied from test-vaults-config.

### Testing Considerations

 - [x] launch a local-chain to test
   - [x] audit vats
 - [ ] try upgrading using the docker test harness

we have unit tests that check such config files, including some static analysis of all the code referenced within:

```
  ✔ Bootstrap SwingSet config file syntax
  ✔ bootstrap permit visualization snapshot (212ms)
  ✔ no test-only code is in production configs (327ms)
    ℹ checking config decentral-main-vaults-config.json
    ℹ decentral-main-vaults-config.json : check bundle: agoricNames vat-agoricNames.js
    ℹ decentral-main-vaults-config.json : check bundle: bank vat-bank.js
    ℹ decentral-main-vaults-config.json : check bundle: board vat-board.js
    ℹ decentral-main-vaults-config.json : check bundle: bridge vat-bridge.js
    ℹ decentral-main-vaults-config.json : check bundle: centralSupply centralSupply.js
    ℹ decentral-main-vaults-config.json : check bundle: mintHolder mintHolder.js
    ℹ decentral-main-vaults-config.json : check bundle: priceAuthority vat-priceAuthority.js
    ℹ decentral-main-vaults-config.json : check bundle: provisionPool provisionPool.js
    ℹ decentral-main-vaults-config.json : check bundle: provisioning vat-provisioning.js
    ℹ decentral-main-vaults-config.json : check bundle: walletFactory walletFactory.js
    ℹ decentral-main-vaults-config.json : check bundle: zcf contractFacet.js
    ℹ decentral-main-vaults-config.json : check bundle: zoe vat-zoe.js
    ℹ checking config decentral-test-vaults-config.json
  ✔ no test-only code is in production proposals (444ms)
    ℹ checking config decentral-main-vaults-config.json
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal0_behaviors startWalletFactory.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal0_0 walletFactory.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal0_1 provisionPool.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_behaviors core-proposal.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_0 binaryVoteCounter.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_1 committee.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_2 contractGovernor.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_3 auctioneer.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_4 feeDistributor.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_5 assetReserve.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal1_6 vaultFactory.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal2_behaviors addAssetToVault.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal2_0 scaledPriceAuthority.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal3_behaviors startPSM.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal3_0 psm.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal3_1 mintHolder.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal10_behaviors price-feed-proposal.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal10_0 fluxAggregatorContract.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal11_behaviors committee-proposal.js
    ℹ decentral-main-vaults-config.json : check bundle: coreProposal11_0 econCommitteeCharter.js
```

### CODE REVIEW

**Comment by dckc** [comment]
@arirubinstein and @JimLarson point out that the upgrade handler should cite this file

https://github.com/Agoric/agoric-sdk/blob/448d8276e893d75d24394b56e80f2f5fd8d7c54b/golang/cosmos/app/app.go#L784

_going back to draft_

File: packages/vats/decentral-main-vaults-config.json [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        }
+      ]
+    },
+    {
+      "module": "@agoric/inter-protocol/scripts/add-collateral-core.js",
+      "entrypoint": "psmProposalBuilder",
+      "args": [
+        {
+          "anchorOptions": {
+            "denom": "ibc/295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F",
```
Comment (ivanlei) [inline packages/vats/decentral-main-vaults-config.json:42]:
```
➜  bin git:(7311-prod-config) ✗ ./agd --node=https://main.rpc.agoric.net:443 query ibc-transfer denom-trace 295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F
denom_trace:
  base_denom: uusdc
  path: transfer/channel-9
```

File: packages/vats/decentral-main-vaults-config.json [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+          "anchorOptions": {
+            "denom": "ibc/3D5291C23D776C3AA7A7ABB34C7B023193ECD2BC42EA19D3165B2CF9652117E7",
+            "decimalPlaces": 18,
+            "keyword": "DAI_grv",
+            "proposedName": "DAI"
+          }
+        }
+      ]
+    },
+    {
```
Comment (ivanlei) [inline packages/vats/decentral-main-vaults-config.json:120]: This stanza should be removed.

File: packages/vats/decentral-main-vaults-config.json [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            "POLL_INTERVAL": 30,
+            "maxSubmissionCount": 1000,
+            "minSubmissionCount": 3,
+            "restartDelay": 1,
+            "timeout": 10,
+            "minSubmissionValue": 1,
+            "maxSubmissionValue": 9007199254740991
+          },
+          "AGORIC_INSTANCE_NAME": "ATOM-USD price feed",
+          "oracleAddresses": [
```
Comment (ivanlei) [inline packages/vats/decentral-main-vaults-config.json:135]: Seemingly there should be only 5 addresses here per comments in https://github.com/Agoric/agoric-sdk/issues/7431

**Review by ivanlei — CHANGES_REQUESTED** [review]
Thanks for the walkthrough of this file. We validated denoms and channels for IBC and talked through the purpose of proposal. 

We will still need some way to ensure vault can not be created until a governance action enables them.

**Comment by ivanlei** [comment]
Based on conversation, can we please ensure through manual testing:
- this bootstrap config works in the docker upgrade tests
- once the chain this creates has settled, can we enumerate every vat and account for what it is and do we want it.

**Comment by dckc** [comment]
## local-chain test confirms 0 mint limit

```
$ mkdir /tmp/dapp1; cd /tmp/dapp1; agoric init config-test
$ cd config-test
$ CHAIN_BOOTSTRAP_VAT_CONFIG=@agoric/vats/decentral-main-vaults-config.json agoric start local-chain --verbose --reset
```

and while that's running, verify 0 mintlimit:
```
$ agd query vstorage data published.vaultFactory.managers.manager0.governance
value: '...{\\\"DebtLimit\\\":{\\\"type\\\":\\\"amount\\\",\\\"value\\\":{\\\"brand\\\":\\\"$0.Alleged:  IST brand\\\",\\\"value\\\":\\\"+0\\\"}}...
```

**Comment by dckc** [comment]
## vat inventory: 48 vats match 45 in unit test

These 48 vats are essentially the same as the 45 vats in our unit test:

https://github.com/Agoric/agoric-sdk/blob/108ae3b328238933750c1f06d094e9b5d6e2b3ac/packages/vats/test/bootstrapTests/test-vaults-upgrade.js#L334-L335

The difference is that there are 6 PSMs (3 vats each) here vs. 5 PSMs in test-vaults-config (- DAI x 2 + AUSD).

```
bash,9693 --init-file /usr/share/codium/resources/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
  `-node,211127 /home/connolly/projects/agoric-sdk/packages/agoric-cli/bin/agoric --sdk start local-chain --verbose --reset
      `-node,211358 /home/connolly/projects/agoric-sdk/packages/cosmic-swingset/bin/ag-chain-cosmos start --home=_agstate/agoric-servers/local-chain-26657
          |-xsnap-worker,212166 v4:vattp -r @8:v4-4
          |-xsnap-worker,212167 v2:vatAdmin -r @8:v2-4
          |-xsnap-worker,212206 v5:timer -r @8:v5-4
          |-xsnap-worker,212240 v8:priceAuthority -r @8:v8-4
          |-xsnap-worker,212243 v11:provisioning -r @8:v11-4
          |-xsnap-worker,212305 v13:zcf-mintHolder-BLD -r @8:v13-4
          |-xsnap-worker,212307 v14:bank -r @8:v14-4
          |-xsnap-worker,212594 v6:agoricNames -r @8:v6-206
          |-xsnap-worker,212715 v15:zcf-b1-eda3a-econCommitteeCharter -r @8:v15-4
          |-xsnap-worker,212721 v16:zcf-mintHolder-ATOM -r @8:v16-4
          |-xsnap-worker,212722 v17:zcf-mintHolder-USDC_axl -r @8:v17-4
          |-xsnap-worker,212723 v18:zcf-mintHolder-USDC_grv -r @8:v18-4
          |-xsnap-worker,212724 v19:zcf-mintHolder-USDT_axl -r @8:v19-4
          |-xsnap-worker,212725 v20:zcf-mintHolder-USDT_grv -r @8:v20-4
          |-xsnap-worker,212754 v21:zcf-mintHolder-DAI_axl -r @8:v21-4
          |-xsnap-worker,212755 v22:zcf-mintHolder-DAI_grv -r @8:v22-4
          |-xsnap-worker,212783 v23:zcf-b1-b4f77-feeDistributor -r @8:v23-4
          |-xsnap-worker,212819 v24:zcf-b1-3f09f-economicCommittee -r @8:v24-4
          |-xsnap-worker,212820 v25:zcf-b1-42a80-walletFactory -r @8:v25-4
          |-xsnap-worker,212889 v26:zcf-b1-2f0d6-ATOM-USD_price_feed-governor -r @8:v26-4
          |-xsnap-worker,212972 v27:zcf-b1-2f0d6-reserve.governor -r @8:v27-4
          |-xsnap-worker,212974 v28:zcf-b1-127f6-ATOM-USD_price_feed -r @8:v28-4
          |-xsnap-worker,213186 v29:zcf-b1-2f0d6-provisionPool-governor -r @8:v29-4
          |-xsnap-worker,213187 v30:zcf-b1-2f0d6-psm-IST-USDC_axl.governor -r @8:v30-4
          |-xsnap-worker,213262 v31:zcf-b1-2f0d6-psm-IST-USDC_grv.governor -r @8:v31-4
          |-xsnap-worker,213263 v32:zcf-b1-2f0d6-psm-IST-USDT_axl.governor -r @8:v32-4
          |-xsnap-worker,213264 v33:zcf-b1-2f0d6-psm-IST-USDT_grv.governor -r @8:v33-4
          |-xsnap-worker,213292 v34:zcf-b1-2f0d6-psm-IST-DAI_axl.governor -r @8:v34-4
          |-xsnap-worker,213297 v35:zcf-b1-2f0d6-psm-IST-DAI_grv.governor -r @8:v35-4
          |-xsnap-worker,213298 v36:zcf-b1-af0de-reserve -r @8:v36-4
          |-xsnap-worker,213306 v37:zcf-b1-33d37-provisionPool -r @8:v37-4
          |-xsnap-worker,213600 v38:zcf-b1-24d83-psm-IST-USDC_axl -r @8:v38-4
          |-xsnap-worker,213605 v39:zcf-b1-24d83-psm-IST-USDC_grv -r @8:v39-4
          |-xsnap-worker,213622 v40:zcf-b1-24d83-psm-IST-USDT_axl -r @8:v40-4
          |-xsnap-worker,213717 v41:zcf-b1-24d83-psm-IST-USDT_grv -r @8:v41-4
          |-xsnap-worker,213737 v42:zcf-b1-24d83-psm-IST-DAI_axl -r @8:v42-4
          |-xsnap-worker,213782 v43:zcf-b1-24d83-psm-IST-DAI_grv -r @8:v43-4
          |-xsnap-worker,213815 v9:zoe -r @8:v9-408
          |-xsnap-worker,213816 v44:zcf-b1-2f0d6-auctioneer.governor -r @8:v44-4
          |-xsnap-worker,214015 v45:zcf-b1-27bc4-auctioneer -r @8:v45-4
          |-xsnap-worker,214048 v46:zcf-b1-620bb-scaledPriceAuthority-ATOM -r @8:v46-4
          |-xsnap-worker,214083 v47:zcf-b1-2f0d6-vaultFactory.governor -r @8:v47-4
          |-xsnap-worker,214241 v48:zcf-b1-00373-vaultFactory -r @8:v48-4
          |-xsnap-worker,214245 v7:board -r @8:v7-206
          |-xsnap-worker,214248 v10:bridge -r @8:v10-206
          `-xsnap-worker,214249 v1:bootstrap -r @8:v1-610
```

made with `pstree -a --hide-threads -l -p 9693 >vaults-vats.txt`

File: packages/vats/decentral-main-vaults-config.json (hunk 2 above)
Comment (dckc) [reply packages/vats/decentral-main-vaults-config.json:120]: done

File: packages/vats/decentral-main-vaults-config.json (hunk 3 above)
Comment (dckc) [reply packages/vats/decentral-main-vaults-config.json:135]: done

**Review by ivanlei — APPROVED** [review]
LGTM 🚀

**Pull request merged**